src/inventory_ui.cpp

```diff
@@ -910,7 +915,21 @@ item_location game::inv_map_splice(
                     }
                 }
             }
-            inv_s.make_item_list( slices.back(), pos == g->u.pos() ? &ground_cat : &nearby_cat );
```

Comment:
The call to inv_s.make_item_list was not deleted. The ternary conditional expression `?` was simply expanded into a full blown `if` below.

> How does this differ from the previous implementation and why are the above changes required?

The mentioned loop was mostly unchanged except for the addition of `match_items_found = true;`. The removal of this line was required to expand the false block of `pos == g->u.pos()`.